design-proposals/vpro-device.md

```diff
@@ -0,0 +1,196 @@
+# Design Proposal: vPRO/AMT/ISM devices activation
+
+Author(s): Edge Infrastructure Manager Team
+
+Last updated: 05/12/2025
+
+## Abstract
+
+vPRO/Active Management Technology (AMT)/Intel Standard Manageability (ISM) needs to be explicitily activated and
+configured in the devices before to be available in the Edge Orchestrator.
+
+[Open Device Management Toolkit](https://device-management-toolkit.github.io/docs/2.27/GetStarted/overview/) (Open DMT)
+provides an open source stack through which is possible to manage vPRO enabled devices.
+
+![EMF Stack](./images/specs/stack.svg)
+
+This document describes the design proposal for adding the Remote Provisioning Client (RPC) component of DMT stack into
+the EN sw and propose a design on how to manage the device activation during the EN journey.
+
+## Proposal
+
+Remote Provisioning Client (RPC) is launched during the provisioning flow (uOS stage).
+
+vPRO (AMT & ISM) capable Core devices need to have the `rpc-go` utility embedded in the uOS image (the so called
+Tink-EMT image - replacement of HookOS), as this is required to activate AMT. This also requires the `heci` driver to
+be included in the kernel as the `rpc-go` utility communicates with CSME over PCIe/HECI in to activate the device.
+
+`Local Manageability Service` (LMS) must to be included as it is still required to enable the communication between RPC
```

Comment:
this is installed as one of our agents ?

Reply:
No this a dep needed by RPC (a mix of user space, kernel modules, etc)